opts/opts.go

```diff
@@ -292,6 +293,77 @@ func (o *NamedMapOpts) Name() string {
 	return o.name
 }
 
+// SetOpts holds a map of values and a validation function.
+type SetOpts struct {
+	values map[string]bool
+}
+
+// Set validates if needed the input value and add it to the
+// internal map, by splitting on '='.
+func (opts *SetOpts) Set(value string) error {
+	k, v, found := strings.Cut(value, "=")
+	var isSet bool
+	if !found {
+		isSet = true
+		k = value
+	} else {
+		var err error
+		isSet, err = strconv.ParseBool(v)
+		if err != nil {
+			return err
+		}
+	}
+	opts.values[k] = isSet
+	return nil
+}
+
+// GetAll returns the values of MapOpts as a map.
+func (opts *SetOpts) GetAll() map[string]bool {
+	return opts.values
+}
+
+func (opts *SetOpts) String() string {
+	return fmt.Sprintf("%v", opts.values)
+}
+
+// Type returns a string name for this Option type
+func (opts *SetOpts) Type() string {
+	return "map"
+}
+
+// NewSetOpts creates a new SetOpts with the specified set of values as a map of string to bool.
+func NewSetOpts(values map[string]bool) *SetOpts {
+	if values == nil {
+		values = make(map[string]bool)
+	}
+	return &SetOpts{
+		values: values,
+	}
+}
```

Comment:
Looks like this one is only used internally to help `NamedSetOpts` - should we combine them? (I'm trying to reduce the public API of this `opts` package - we could even consider making the new one un-exported in `daemon/config` or somewhere in `internal/` or `daemon/internal`.